public/app/features/alerting/unified/Silences.tsx

```diff
@@ -31,33 +38,65 @@ const Silences: FC = () => {
     };
   }, [alertManagerSourceName, dispatch]);
 
+  const { result, loading, error } = silences[alertManagerSourceName] || initialAsyncRequestState;
+  const getSilenceById = useCallback((id: string) => result && result.find((silence) => silence.id === id), [result]);
+
   if (!alertManagerSourceName) {
     return <Redirect to="/alerting/silences" />;
   }
-  const { result, loading, error } = silences[alertManagerSourceName] || initialAsyncRequestState;
 
   return (
     <AlertingPageWrapper pageId="silences">
       <Field label="Choose alert manager">
         <AlertManagerPicker current={alertManagerSourceName} onChange={setAlertManagerSourceName} />
       </Field>
-      <br />
-      <br />
+      {isRoot && (
+        <Link href={`${config.appSubUrl ?? ''}/alerting/silence/new`}>
```

Comment:
```suggestion
        <Link href={makeAMLink('/alerting/silence/new', alertManagerSourceName)}>
```

We should always include `alertManagerSourceName` in the link to prevent browser from loading it from store and doing a redirect. messes with back button and such